pkg/admission/validator_test.go

```diff
@@ -0,0 +1,226 @@
+package admission
+
+import (
+	"bytes"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/lithammer/dedent"
+	"github.com/stretchr/testify/assert"
+	admission "k8s.io/api/admission/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	v1alpha1 "sigs.k8s.io/service-apis/apis/v1alpha1"
+)
+
+var decoder = codecs.UniversalDeserializer()
+
+func TestServeHTTPBasic(t *testing.T) {
+	assert := assert.New(t)
+	res := httptest.NewRecorder()
+	handler := http.HandlerFunc(ServeHTTP)
+	req, err := http.NewRequest("POST", "", nil)
+	assert.Nil(err)
+	handler.ServeHTTP(res, req)
+	assert.Equal(400, res.Code)
+	assert.Equal("admission review object is missing\n",
+		res.Body.String())
+}
+
+func TestValidationWebhook(t *testing.T) {
+	for _, apiVersion := range []string{
+		"admission.k8s.io/v1beta1",
+		"admission.k8s.io/v1",
+	} {
+		for _, tt := range []struct {
+			name    string
+			reqBody string
+
+			wantRespCode        int
+			wantSuccessResponse admission.AdmissionResponse
+			wantFailureMessage  string
+		}{
+			{
+				name:               "request with present empty body",
+				wantRespCode:       http.StatusBadRequest,
+				wantFailureMessage: "unexpected end of JSON input\n",
+			},
+			{
+				name: "one valid requestMirror in httpRoute",
+				reqBody: dedent.Dedent(`{
+						"kind": "AdmissionReview",
+						"apiVersion": "` + apiVersion + `",
+						"request": {
+							"uid": "b2df61dd-ab5b-4cb4-9be0-878533c83892",
+							"resource": {
+								"group": "networking.x-k8s.io",
+								"version": "v1alpha1",
+								"resource": "httproutes"
+							},
+							"object": {
+								"apiVersion": "networking.x-k8s.io/v1alpha1",
+								"kind": "HTTPRoute"
+							},
+						"operation": "CREATE"
+						}
+					}`),
+				wantRespCode: http.StatusOK,
+				wantSuccessResponse: admission.AdmissionResponse{
+					UID:     "b2df61dd-ab5b-4cb4-9be0-878533c83892",
+					Allowed: true,
+					Result:  &metav1.Status{},
+				},
+			},
+		} {
+			t.Run(fmt.Sprintf("%s/%s", apiVersion, tt.name), func(t *testing.T) {
+				assert := assert.New(t)
+				res := httptest.NewRecorder()
+				handler := http.HandlerFunc(ServeHTTP)
+
+				// send request
+				req, err := http.NewRequest("POST", "", bytes.NewBuffer([]byte(tt.reqBody)))
+				assert.Nil(err)
+				handler.ServeHTTP(res, req)
+
+				// check response assertions
+				assert.Equal(tt.wantRespCode, res.Code)
+				fmt.Println(res.Body.String())
```

Comment:
Was this just in here temporarily for debugging?

Reply:
Most likely, will remove